### ISSUE TITLE
cause use recycled bitmap then crash

### ISSUE THREAD

**Issue by ChanNagihong** [davemorrissey/subsampling-scale-image-view#377]
boolean initialized = false;
Bitmap onCreateBitmap;
Bitmap onResumeBitmap;

void onCreate() {
    onCreateBitmap = readBitmap();
    putBitmapIntoPool(onCreateBitmap);
    show(onCreateBitmap);
}

void onResume() {
    if(initialized) {
        onResumeBitmap = readBitmapFromPool(); // onResumeBitmap == onCreateBitmap return true
        show(onResumeBitmap);
    } else {
        initialized = !initialized;
    }
}

void show(Bitmap bitmap) {
    photoView.setImage(ImageSource.bitmap(bitmap));       
}

**Comment by davemorrissey** [comment]
You haven't provided the stack trace or completed the requested template, but I'm guessing you can fix this with `ImageSource.cachedBitmap(bitmap)`

**Comment by ChanNagihong** [comment]
I tried and it works, thanks a lot

**Comment by blinkmacalahan** [comment]
I wanted to add some additional info for anyone else having a similar issue as me in case the noted solution above works for them as well. I'm using this view inside RecyclerView (with PageSnapHelper) and loading the bitmap via Glide. After the RecyclerView item was getting recycled, I started seeing:


`
Bitmap  : Called getAllocationByteCount() on a recycle()'d bitmap! This is undefined behavior!
Bitmap  : Called getConfig() on a recycle()'d bitmap! This is undefined behavior!
Bitmap  : Called getWidth() on a recycle()'d bitmap! This is undefined behavior!
Bitmap  : Called getHeight() on a recycle()'d bitmap! This is undefined behavior!
`

After re-using a couple of recycled views, the app would crash without a stack trace, but I noticed this error in logcat:
`
03-11 15:35:31.366 19824 19824 F Bitmap  : Error, cannot access an invalid/free'd bitmap here!
`

Using `ImageSource.cachedBitmap(bitmap)` fixed the crash. If you read the javadocs for `cachedBitmap(bitmap)` it says 

> Provide a loaded and cached bitmap for display. This bitmap will not be recycled when it is no
    longer needed. Use this method if you loaded the bitmap with an image loader such as Picasso
     or Volley.

Works for Glide as well!
